src/compiler/checker.ts

```diff
@@ -15459,6 +15463,39 @@ namespace ts {
                     }
                 }
             }
+
+            function getUnImplementedAncestorMembers(type: InterfaceType): SymbolTable {
+                const results: SymbolTable = {};
+
+                const baseTypes = getAbstractAncestorTypes(type);
+                for (const bt of baseTypes) {
+                    const abstractMembers = getNamedMembers(bt.symbol.members).filter(isSymbolAbstract);
+                    for (const am of abstractMembers) {
+                        results[am.name] = am;
+                    }
+                }
+
+                for (const member of getPropertiesOfType(type).filter(m => !isSymbolAbstract(m))) {
+                    delete results[member.name];
+                }
+
+                return results;
+            }
+
+            // TODO: Are we allowed to use a generator here?
+            function getAbstractAncestorTypes(type: InterfaceType): Type[] {
+                let ancestors: Type[] = [];
+
+                for (const bt of getBaseTypes(type).filter(t => isSymbolAbstract(t.symbol))) {

```

Comment:
Using `getBaseTypes` is wasteful because you can only extend from one class, and only classes can be abstract anyway